cdap-app-fabric/src/test/java/co/cask/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -234,9 +268,14 @@ public void testRunScheduledJobs() throws Exception {
     int runs2 = getRuns(WORKFLOW_2);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_1);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_2);
+
     publishNotification(dataEventTopic, WORKFLOW_1, AppWithFrequentScheduledWorkflows.DATASET_NAME1);
+    MessageId messageId = waitUntilProcessed(dataEventTopic, null);
```

Comment:
I updated the unit-test to use publish timestamp instead of just comparing the message id changed. Also added the validation of the job queue instead of just checking if the program is not triggered.